dev/proglang/src/main.clj

```diff
@@ -233,18 +234,18 @@
     [:equal left right] (mdo [left (m-eval left)
                               right (m-eval right)
                               _ [:pure [:bool (= left right)]]])
-    [:if condition if-true if-false] (mdo [condition (m-eval condition)
-                                           _ (if (contains? #{[:bool false] [:int 0]} condition)
-                                               (m-eval (cons :S if-false))
-                                               (m-eval (cons :S if-true)))])
+    [:if condition if-true & [if-false]] (mdo [condition (m-eval condition)
+                                               _ (if (contains? #{[:bool false] [:int 0]} condition)
+                                                   (m-eval (cons :S if-false))
+                                                   (m-eval (cons :S if-true)))])
     [:print expr] (mdo [v (m-eval expr)
                         _ [:print v]])
-    [:S head & tail] (cond (nil? head) [:pure nil]
-                           (empty? tail) (m-eval head)
-                           :else (mdo [[ret? v] (m-eval head)
-                                       _ (if (= :return ret?)
-                                           [:pure [:return v]]
-                                           (m-eval (cons :S tail)))]))))
+    [:S] [:pure nil]
+    [:S head] (m-eval head)
```

Comment:
This one is not actually needed; if we remove this line, we use the next one, and it's functionally the same. However, I find it a bit clearer this way. I might change my mind at some point. Either way, I like that `vatch` supports this.